Deploy ganesha daemons with vstart

varshar16 opened the pull request:
<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->

To create Ganesha daemons with vstart following packages needs to be installed:
nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace

This patch is tested on fedora 30 system with packages version  2.7.6-2.

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

jtlayton commented on src/vstart.sh:
```diff
@@ -1041,6 +1047,90 @@ EOF
+        if !(firewall-cmd --list-services | grep nfs) ; then
+            prun firewall-cmd --add-service=nfs
+            prun firewall-cmd --reload
+        fi
```
Given that you aren't running ganesha on standard NFS port, do you need to monkey with the firewall?

varshar16 replied:
I did not do any modification to the firewall.

jtlayton replied:
*blink*

firewall-cmd does exactly that. You're adding the NFS "service" to it here and reloading it. I don't think that's necessary since you're running this on a random port.

varshar16 replied:
Sorry, I meant apart from firewall-cmd.

jtlayton commented on src/vstart.sh:
```diff
@@ -1243,6 +1333,12 @@ osd pool create ec erasure ec-profile
+if [ $GANESHA_DAEMON_NUM -gt 0 ]; then
+    echo "Starting ganesha"
+    start_ganesha
+fi
```
It might be good to also print out the ports after starting the daemons. NFS clients will need to know which port to talk to since it's not sitting on a standard port.

varshar16 replied:
I have updated the patch to print the ports.

varshar16 commented on src/vstart.sh:
```diff
@@ -1041,6 +1047,87 @@ EOF
+        ganesha=$(($ganesha + 1))
+        ganesha_dir="$CEPH_DEV_DIR/ganesha.$name"
+
+        echo "Starting ganesha on port: $port"
```
Done

varshar16 commented on src/vstart.sh:
```diff
@@ -1041,6 +1047,87 @@ EOF
+EOF
+        prun $CEPH_BIN/rados lspools
+
+        prun ganesha-rados-grace -p cephfs.a.meta -n ganesha add $name
```
I have created new pool for ganesha.

varshar16 commented:
jenkins retest this please

varshar16 commented:
Disabled root squashing.

varshar16 commented:
Thank you for the reviews Jeff and Ramana.

batrick commented:
> nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace

Where are you testing this PR? I don't find these packages available on CentOS 8 or Ubuntu 16.04.

varshar16 commented:
> > nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace
> 
> Where are you testing this PR? I don't find these packages available on CentOS 8 or Ubuntu 16.04.

I tested them on Fedora 30 and package version is  2.7.6-2. For CentOS 8 the packages are available [here](https://download.nfs-ganesha.org/2.7/2.7.6/CentOS/) and Ubuntu 16.04 check [here](https://launchpad.net/~nfs-ganesha/+archive/ubuntu/nfs-ganesha-2.7). On Ubuntu I only found nfs-ganesha package.

batrick commented:
> > > nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace
> > 
> > 
> > Where are you testing this PR? I don't find these packages available on CentOS 8 or Ubuntu 16.04.
> 
> I tested them on Fedora 30 and package version is 2.7.6-2. For CentOS 8 the packages are available [here](https://download.nfs-ganesha.org/2.7/2.7.6/CentOS/) and Ubuntu 16.04 check [here](https://launchpad.net/~nfs-ganesha/+archive/ubuntu/nfs-ganesha-2.7). On Ubuntu I only found nfs-ganesha package.

Okay, so we need the upstream repo. Can you add instructions for doing that to vstart in a comment so others know what to setup/install.

varshar16 commented:
> > > > nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace
> > > 
> > > 
> > > Where are you testing this PR? I don't find these packages available on CentOS 8 or Ubuntu 16.04.
> > 
> > 
> > I tested them on Fedora 30 and package version is 2.7.6-2. For CentOS 8 the packages are available [here](https://download.nfs-ganesha.org/2.7/2.7.6/CentOS/) and Ubuntu 16.04 check [here](https://launchpad.net/~nfs-ganesha/+archive/ubuntu/nfs-ganesha-2.7). On Ubuntu I only found nfs-ganesha package.
> 
> Okay, so we need the upstream repo. Can you add instructions for doing that to vstart in a comment so others know what to setup/install.

I have updated the initial PR comment. Let me know if it is fine.

batrick commented:
> > > > > nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace
> > > > 
> > > > 
> > > > Where are you testing this PR? I don't find these packages available on CentOS 8 or Ubuntu 16.04.
> > > 
> > > 
> > > I tested them on Fedora 30 and package version is 2.7.6-2. For CentOS 8 the packages are available [here](https://download.nfs-ganesha.org/2.7/2.7.6/CentOS/) and Ubuntu 16.04 check [here](https://launchpad.net/~nfs-ganesha/+archive/ubuntu/nfs-ganesha-2.7). On Ubuntu I only found nfs-ganesha package.
> > 
> > 
> > Okay, so we need the upstream repo. Can you add instructions for doing that to vstart in a comment so others know what to setup/install.
> 
> I have updated the initial PR comment. Let me know if it is fine.

I mean please add instructions to setup the dev repositories https://download.nfs-ganesha.org/2.7/2.7.6/ in order to install the packages. This could be a comment in vstart.sh

varshar16 commented:
> > > > > > nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rados-grace
> > > > > 
> > > > > 
> > > > > Where are you testing this PR? I don't find these packages available on CentOS 8 or Ubuntu 16.04.
> > > > 
> > > > 
> > > > I tested them on Fedora 30 and package version is 2.7.6-2. For CentOS 8 the packages are available [here](https://download.nfs-ganesha.org/2.7/2.7.6/CentOS/) and Ubuntu 16.04 check [here](https://launchpad.net/~nfs-ganesha/+archive/ubuntu/nfs-ganesha-2.7). On Ubuntu I only found nfs-ganesha package.
> > > 
> > > 
> > > Okay, so we need the upstream repo. Can you add instructions for doing that to vstart in a comment so others know what to setup/install.
> > 
> > 
> > I have updated the initial PR comment. Let me know if it is fine.
> 
> I mean please add instructions to setup the dev repositories https://download.nfs-ganesha.org/2.7/2.7.6/ in order to install the packages. This could be a comment in vstart.sh

I have added them. I did not find `nfs-ganesha-rados-grace` and `nfs-ganesha-ceph` packages for Ubuntu 16.04.
